solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/XMLMessageMapper.java

```diff
@@ -55,6 +57,7 @@
 
 	private final ObjectWriter stringSetWriter = OBJECT_MAPPER.writerFor(new TypeReference<Set<String>>(){});
 	private final ObjectReader stringSetReader = OBJECT_MAPPER.readerFor(new TypeReference<Set<String>>(){});
+	private final AtomicBoolean readDeliveryCount = new AtomicBoolean(true);
```

Comment:
Should be called something like `readUnsupportedProperties` to generically capture any property which throws an `UnsupportedOperationException`.

Reply:
Actually, this might need to be refactored to a set of unsupported `SolaceHeaders` to handle this generically.